client/landing/stepper/declarative-flow/start-writing.ts

```diff
@@ -0,0 +1,79 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { START_WRITING_FLOW } from '@automattic/onboarding';
+import { useSelector } from 'react-redux';
+import { recordSubmitStep } from 'calypso/landing/stepper/declarative-flow/internals/analytics/record-submit-step';
+import { redirect } from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/import/util';
+import {
+	AssertConditionResult,
+	AssertConditionState,
+	Flow,
+	ProvidedDependencies,
+} from 'calypso/landing/stepper/declarative-flow/internals/types';
+import { getCurrentUserSiteCount, isUserLoggedIn } from 'calypso/state/current-user/selectors';
+
+const startWriting: Flow = {
+	name: START_WRITING_FLOW,
+	useSteps() {
+		return [
+			{
+				slug: 'site-creation-step',
+				asyncComponent: () => import( './internals/steps-repository/site-creation-step' ),
+			},
+			{
+				slug: 'processing',
+				asyncComponent: () => import( './internals/steps-repository/processing-step' ),
+			},
+		];
+	},
+
+	useStepNavigation( currentStep, navigate ) {
+		const flowName = this.name;
+		function submit( providedDependencies: ProvidedDependencies = {} ) {
+			recordSubmitStep( providedDependencies, '', flowName, currentStep );
+			switch ( currentStep ) {
+				case 'site-creation-step':
+					return navigate( 'processing' );
+				case 'processing': {
+					if ( providedDependencies?.siteSlug ) {
+						return redirect(
+							`https://${ providedDependencies?.siteSlug }/wp-admin/post-new.php?showLaunchpad=true`
```

Comment:
Adding the origin here can help with redirection [here](https://github.com/Automattic/wp-calypso/pull/75649/files#diff-2a5c4248951649b4041f644a42855e4d740aee9b594680695e07eb43bc1bb821R5).
```
const siteOrigin = window.location.origin;
```
```suggestion
							`https://${ providedDependencies?.siteSlug }/wp-admin/post-new.php?showLaunchpad=true&origin=${ siteOrigin }`
```

Reply:
Thanks @candy02058912, that's a great suggestion! Can we move forward with that @DustyReagan?

Reply:
Good idea! ✅